frontend/ui/widget/progresswidget.lua

```diff
@@ -90,8 +92,8 @@ function ProgressWidget:paintTo(bb, x, y)
         local bar_width = (my_size.w-2*self.margin_h)
         local y_pos = y + self.margin_v + self.bordersize
         local bar_height = my_size.h-2*(self.margin_v+self.bordersize)
-        for i=1, #self.ticks do
-            local tick_x = bar_width*(self.ticks[i]/self.last)
+        for i,tick in pairs(self.ticks) do
+            local tick_x = bar_width*((tick-0.5)/self.last)
```

Comment:
I'm... not necessarily a fan of that ;).

And, at the very least, it'll probably require rounding to ensure it doesn't randomly decide to jump a pixel or not ;).

Reply:
Well, either there is a non-existing "page 0" represented in the progress bar, or the last page is missing. If I use the plain page numbers for the non-linear filling, I get an overflow:
![Screenshot_20201107_153402](https://user-images.githubusercontent.com/7943459/98443981-4f160000-210f-11eb-8e3b-ab30c3d7ec39.png)

which I can solve by using `page-1`, but then the ticks are off:
![Screenshot_20201107_153445](https://user-images.githubusercontent.com/7943459/98443983-4fae9680-210f-11eb-80e3-4a6c5f39b436.png)

So with `tick-0.5` the ticks are "midpage", they're still off, but at least they're not at the boundary of any page:
![Screenshot_20201107_153527](https://user-images.githubusercontent.com/7943459/98443984-4fae9680-210f-11eb-873b-8cee4895cc5c.png)

As for rounding, I don't think it's needed. `tick` should be an integer, and even though `tick_x` can jump one pixel, it will always be between the values corresponding to `tick` and `tick-1`. In the case when each page takes less than one pixel, I don't really care which one it is, there's not enough resolution anyway.

Reply:
I'd rather keep the current behavior and just fix the overflow, FWIW ;).

Reply:
To fix the overflow and keep the ticks at the boundary, everything should be moved `-1` page. Then the first page would have no filled progress bar, and the last page would not be 100% filled (which would make sense if the bar means "pages already read"), or the progress bar in divided in `pages-1` chunks, instead of `pages`.

But ok, I'll leave it as is. I don't really use the skim widget enough.

Reply:
> As for rounding, I don't think it's needed

Not rounding what might be passed to the drawing code has been the caused of positionning & fitting issues (even in upper containers far from what you're dealing with) in the past. So, please do if you can :)

I have no real opinion on #6831 because I don't use it that much to jump - I just expect it to be "clean", so my preference among the 3 above screenshots is the first (without the overflow of course :) The 2 others, no matter how correct they may be for the person who gave it some thoughts, will look like a bug for most of the other users :/

Reply:
The first is wronger, because the first page is non-linear, and it appears as the second, it's the same -1 offset. If the first page is non-linear, it should go from 0 to 1, and if the last 2 pages (out of 100) are non-linear, the grey should go from 98 to 100 (even though the pages are 99 and 100).

So, my point is the grey blocks in the 2nd and 3rd example are correct. The ticks in 1st and 2nd are correct, if they are interpreted as marking the end of a page where a chapter starts, and the filled bar is correct if it's interpreted as all pages up to (and including) the current one.

Reply:
My stance on the whole #6381 discussion was that I might be agreeable to modify the way the *bar* is filled in, but that the ticks should very much stick to what they're currently doing.

Reply:
> To fix the overflow and keep the ticks at the boundary, everything should be moved `-1` page. Then the first page would have no filled progress bar, and the last page would not be 100% filled (which would make sense if the bar means "pages already read"), or the progress bar should be divided in `pages-1` chunks, instead of `pages`.

FWIW, I'd probably be amenable to either of those approaches, I think ;).

(Never seeing the bar fully filled on the last page kinda makes sense to me).